File: benchmark/results/python-multilang/run-002/a-mcp/output/billing/services/invoice_repository.py
from typing import Any, Dict, List, Optional
from billing.aggregates.invoice import Invoice

"""InvoiceRepository service."""
class InvoiceRepository:
    def create(self, input: Dict[str, Any]) -> Invoice:
        raise NotImplementedError('not implemented')
    def findById(self, id: str) -> Optional[Invoice]:
        raise NotImplementedError('not implemented')
    def list(self, limit: int) -> List[Invoice]:
        raise NotImplementedError('not implemented')
    def delete(self, id: str) -> None:
        raise NotImplementedError('not implemented')

File: benchmark/results/java-invocation-inline/run-001/b-baseline/output/com/metaengine/demo/billing/services/InvoiceRepository.java
package com.metaengine.demo.billing.services;

import com.metaengine.demo.billing.aggregates.Invoice;
import java.util.List;

/** Persistence gateway for invoices. */
public class InvoiceRepository {
    /** Persist a new invoice from the provided partial input. */
    public Invoice create(Invoice input) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Load an invoice by id. Returns null when not found. */
    public Invoice findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** List up to {@code limit} invoices. */
    public List<Invoice> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    /** Remove an invoice by id. */
    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}
